### DISCUSSION TITLE
Javascript

### DISCUSSION THREAD

**Discussion by ggerganov** [ggml-org/whisper.cpp#309] [Bindings]
General discussion about the `javascript` bindings for `whisper.cpp`

**Current state:** [bindings/javascript](https://github.com/ggerganov/whisper.cpp/tree/master/bindings/javascript)

**Comment by haikyuu** [comment]
If I understand correctly, the performance using the binding is degraded.

Say I have a nodejs app, and I want maximum performance, I'm better off using `child_process` to run whisper.cpp in the cli, right?

**Reply by ggerganov** [reply]
I think the best option for a node.js app is to use `whisper.cpp` as "addon".
See the example here:

https://github.com/ggerganov/whisper.cpp/tree/master/examples/addon.node

**Reply by haikyuu** [reply]
yes, that makes much more sense since node.js plays well with c++ 

Thanks

**Reply by gkostov** [reply]
So do you think it'd be useful to prepare, publish and maintain an NPM package, named maybe `whisper.cpp.node`, which to allow for simpler NPM-style installation of the addon? Sparing the hassle of building it.

There are many similar packages on NPM but they are forks and variations, and what not, and it's hard to tell what their code really does and for how long they will be maintained. Having a package that directly references this code base will make it clear what exactly it is about.
________
P.S.
And thinking again about it, a separate NPM package is likely not needed as the node addon can be built (on-demand, as it'll take to download the repo) as part of the existing `whisper.cpp`)

**Reply by starNGC2237** [reply]
Maybe like [nodejs-whisper](https://github.com/ChetanXpro/nodejs-whisper) or [whisper-node-addon](https://github.com/starNGC2237/whisper-node-addon) ?

**Reply by gkostov** [reply]
Yes, but without distributing repositories or pre-built binaries through NPM. Only what ggml-org/whisper.cpp/tree/master/examples/addon.node gives you, as if you'd download the repo and build it on your machine, but without having to remember urls and build commands, and having to go thorugh a number of directories, etc. I understand it might be usefult to have a dozen of other options in the package but having to update, build and publish a new package for every new whisper version seems quite the hassle and causes packages to go stale. My suggestion in #3625 is to allow easy downloading either the version that relates to the package or the latest master and building it on your machine. There could be more options added (like building with CUDA/CoreML/etc.) but only such that do not go way out of the scope of the addon, that's all.
